stubs/openpyxl/openpyxl/worksheet/worksheet.pyi

```diff
@@ -24,106 +33,106 @@ class Worksheet(_WorkbookChild):
     PAPERSIZE_A5: str
     ORIENTATION_PORTRAIT: str
     ORIENTATION_LANDSCAPE: str
-    def __init__(self, parent, title: Incomplete | None = ...) -> None: ...
+    def __init__(self, parent: Workbook, title: str | None = ...) -> None: ...
     @property
-    def sheet_view(self): ...
+    def sheet_view(self) -> SheetView: ...
     @property
-    def selected_cell(self): ...
+    def selected_cell(self) -> Cell: ...
     @property
-    def active_cell(self): ...
+    def active_cell(self) -> Cell: ...
     @property
-    def page_breaks(self): ...
+    def page_breaks(self) -> tuple[RowBreak, ColBreak]: ...
     @property
-    def show_gridlines(self): ...
+    def show_gridlines(self) -> bool: ...
     @property
-    def show_summary_below(self): ...
+    def show_summary_below(self) -> bool: ...
     @property
-    def show_summary_right(self): ...
+    def show_summary_right(self) -> bool: ...
     @property
-    def freeze_panes(self): ...
+    def freeze_panes(self) -> str | None: ...
     @freeze_panes.setter
     def freeze_panes(self, topLeftCell: Incomplete | None = ...) -> None: ...
-    def cell(self, row, column, value: Incomplete | None = ...): ...
-    def __getitem__(self, key): ...
-    def __setitem__(self, key, value) -> None: ...
-    def __iter__(self): ...
-    def __delitem__(self, key) -> None: ...
+    def cell(self, row: int, column: int, value: str | None = ...) -> Cell: ...
+    def __getitem__(self, key: str | int | slice) -> Cell | tuple[Cell, ...]: ...
+    def __setitem__(self, key: str, value: str) -> None: ...
+    def __iter__(self) -> Iterator[Cell]: ...
+    def __delitem__(self, key: str) -> None: ...
     @property
-    def min_row(self): ...
+    def min_row(self) -> int: ...
     @property
-    def max_row(self): ...
+    def max_row(self) -> int: ...
     @property
-    def min_column(self): ...
+    def min_column(self) -> int: ...
     @property
-    def max_column(self): ...
-    def calculate_dimension(self): ...
+    def max_column(self) -> int: ...
+    def calculate_dimension(self) -> str: ...
     @property
-    def dimensions(self): ...
+    def dimensions(self) -> str: ...
     def iter_rows(
         self,
-        min_row: Incomplete | None = ...,
-        max_row: Incomplete | None = ...,
-        min_col: Incomplete | None = ...,
-        max_col: Incomplete | None = ...,
+        min_row: int | None = ...,
+        max_row: int | None = ...,
+        min_col: int | None = ...,
+        max_col: int | None = ...,
         values_only: bool = ...,
```

Comment:
@AlexWaygood I struggled to implement the @overloads. Kept on getting `Overloaded function signatures 1 and 2 overlap with incompatible return types` errors when using `Literal[True]` and `Literal[False]` for the two possible return types.

Reply:
Yes, the overloads are a little tricky to get right here. I took the liberty of pushing to your branch to add them in 8ed83b4

Reply:
Is forcing the argument to be a keyword argument the only way around those errors?

Reply:
So, the issue here is that the type signature we'd "like" to write would be something like this. Note that each overload has a different return type:

```python
@overload
def iter_rows(
    self,
    min_row: int | None = None,
    max_row: int | None = None,
    min_col: int | None = None,
    max_col: int | None = None,
    values_only: Literal[True],
) -> Generator[tuple[str | float | datetime | None, ...], None, None]: ...
@overload
def iter_rows(
    self,
    min_row: int | None = ...,
    max_row: int | None = ...,
    min_col: int | None = ...,
    max_col: int | None = ...,
    values_only: Literal[False] = False,
) -> Generator[tuple[Cell, ...], None, None]: ...
@overload
def iter_rows(
    self,
    min_row: int | None = None,
    max_row: int | None = None,
    min_col: int | None = None,
    max_col: int | None = None,
    values_only: bool,
) -> Generator[tuple[Cell | str | float | datetime | None, ...], None, None]: ...
```

That would clearly spell out to the type checker that there are three distinct things that could happen here:
- The user could pass in a value that's known to be `Literal[True]` in, in which case the first overload is selected
- The user could either not pass in a value for that parameter, or pass a value that's known to be `Literal[False]` in, in which case the second overload is selected
- The user could pass in a value that's known to be a `boolean` of some kind, but isn't known to be either `Literal[True]` or `Literal[False]`, in which case the third parameter is selected.

Importantly, the second overload is the only one here that has a default value given for the `values_only` parameter, because we want to make sure that the type checker always selects this overload if the user doesn't pass in a value for this parameter.

_Unfortunately_, two of those^ overloads result in `SyntaxError`s when parsed by Python, because you can't have a parameter with a default value preceding a parameter without a default value in a function signature. That means that we have to expand out the first overload into two overloads, in order to make sure that we're still allowing all valid function calls:

```python
@overload
def iter_rows(
    self,
    min_row: int | None,
    max_row: int | None,
    min_col: int | None,
    max_col: int | None,
    values_only: Literal[True],
) -> Generator[tuple[str | float | datetime | None, ...], None, None]: ...
@overload
def iter_rows(
    self,
    min_row: int | None = None,
    max_row: int | None = None,
    min_col: int | None = None,
    max_col: int | None = None,
    *,
    values_only: Literal[True],
) -> Generator[tuple[str | float | datetime | None, ...], None, None]: ...
```

And we have to do the same with the third overload, which has the same issue. That means that we end up with the five overloads that I added in https://github.com/python/typeshed/commit/8ed83b411545cafe16c5d2c9d0089dd6f12e1f8a